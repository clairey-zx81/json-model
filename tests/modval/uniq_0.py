# Generated for model: ./uniq_0.model.json
#
# Generated by jmc (JSON Model Compiler) (https://github.com/clairey-zx81/json-model)
#

from typing import Callable
import re as re
import datetime
import urllib.parse

type Jsonable = None|bool|int|float|str|list[Jsonable]|dict[str, Jsonable]
type CheckFun = Callable[[Jsonable, str], bool]
type PropMap = dict[str, CheckFun]
type TagMap = dict[None|bool|float|int|str, CheckFun]


# define "$" ($)
def json_model_0(value: Jsonable, path: str) -> bool:
    # $
    # $.'@'
    result = isinstance(value, list)
    if result:
        assert isinstance(value, list)  # pyright helper
        for array_0_idx, array_0_item in enumerate(value):
            # $.'@'.0
            result = isinstance(array_0_item, int) and not isinstance(array_0_item, bool) and array_0_item >= 1
            if not result: break
    result &= len(set(value)) == len(value)
    return result

# entry function check_model
def check_model(value: Jsonable, path: str = "$") -> bool:
    return json_model_0(value, path)


