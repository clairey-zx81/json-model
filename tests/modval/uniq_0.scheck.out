uniq_0.0.true.json: PASS
uniq_0.1.true.json: PASS
uniq_0.2.true.json: PASS
uniq_0.3.false.json: FAIL ['invalid type or constraints at $ [$]']
uniq_0.4.false.json: FAIL ["not a 1 int at $.0 [$.'@'.0]", "not array or unexpected array at $ [$.'@']", 'invalid type or constraints at $ [$]']
